src/rdo.rs

```diff
@@ -328,20 +330,27 @@ pub fn rdo_mode_decision(
           false
         );
 
-        if rd < best_rd {
-          best_rd = rd;
-          best_mode_luma = luma_mode;
-          best_mode_chroma = chroma_mode;
-          best_cfl_params = cfl;
-          best_ref_frame = ref_frame;
-          best_mv = mv;
-          best_skip = skip;
-        }
-
         cw.rollback(&cw_checkpoint);
-      }
-    }
-  }
+
+        (rd, luma_mode, chroma_mode, cfl, ref_frame, mv, skip)
+      }).fold((best_rd, best_mode_luma, best_mode_chroma, best_cfl_params, best_ref_frame, best_mv, best_skip),
+        |best, cur| {
+        if cur.0 < best.0 {
+          cur
+        } else {
+          best
+        }
+      });
+
+      best_rd = best.0;
```

Comment:
Looks like my guess of best being a tuple or vector seems correct, since each field (best_rd, best_mode_luma, ... ) is indexed by .number...  What a rust!

Reply:
Notice that then you can spare all this noise using `min_by()` :)